backend/src/apiserver/server/visualization_server.go

```diff
@@ -78,11 +79,8 @@ func (s *VisualizationServer) validateCreateVisualizationRequest(request *go_cli
 // It returns the generated HTML as a string and any error that is encountered.
 func (s *VisualizationServer) generateVisualizationFromRequest(request *go_client.CreateVisualizationRequest) ([]byte, error) {
 	serviceURL := s.getVisualizationServiceURL(request)
-	if !isVisualizationServiceAlive(serviceURL) {
-		return nil, util.NewInternalServerError(
-			fmt.Errorf("service not available"),
-			"Service not available",
-		)
+	if err := isVisualizationServiceAlive(serviceURL); err != nil {
+		return nil, err
```

Comment:
nit: instead of returning err directly, suggest using util.Wrap on it.

The extra layer of error message provide more context.

It answers:
Why did we verify visualization server is alive?
It's because we are trying to generate a visualization.

Reply:
Right, added one layer of `util.Wrap` on it.